File: src/test/java/io/github/subiyacryolite/jds/JdsDbSqliteImplementation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package io.github.subiyacryolite.jds;

import org.sqlite.SQLiteConfig;

import java.io.File;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

/**
 *
 * @author indana
 */
public class JdsDbSqliteImplementation extends JdsDbSqlite {

    @Override
    public Connection getConnection() throws ClassNotFoundException, SQLException {
        String url = "jdbc:sqlite:" + getDatabaseFile();
        SQLiteConfig sqLiteConfig = new SQLiteConfig();
        sqLiteConfig.enforceForeignKeys(true); //You must enable foreign keys in SQLite
        Class.forName("org.sqlite.JDBC");
        return DriverManager.getConnection(url, sqLiteConfig.toProperties());
    }

    public String getDatabaseFile() {
        File path = new File(System.getProperty("user.home") + File.separator + ".jdstest" + File.separator + "jds.db");
        if (!path.exists()) {
            File directory = path.getParentFile();
            if (!directory.exists()) {
                directory.mkdirs();
            }
        }
        String absolutePath = path.getAbsolutePath();
        return absolutePath;
    }
}

File: src/test/java/connectivity/JdsDbSqliteImplementation.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package connectivity

import io.github.subiyacryolite.jds.JdsDbSqlite
import org.sqlite.SQLiteConfig
import java.io.File
import java.sql.Connection
import java.sql.DriverManager

/**
 *
 * @author indana
 */
class JdsDbSqliteImplementation : JdsDbSqlite() {

    private val fileLocation: String
        get() {
            val path = File(System.getProperty("user.home") + File.separator + ".jdstest" + File.separator + "jds.db")
            if (!path.exists()) {
                val directory = path.parentFile
                if (!directory.exists()) {
                    directory.mkdirs()
                }
            }
            return path.absolutePath
        }

    override val connection: Connection
        get () {
            val url = "jdbc:sqlite:$fileLocation"
            val sqLiteConfig = SQLiteConfig()
            sqLiteConfig.enforceForeignKeys(true) //You must enable foreign keys in SQLite
            Class.forName("org.sqlite.JDBC")
            return DriverManager.getConnection(url, sqLiteConfig.toProperties())
        }
}
